test/bpf_tests/bpf_test.go

```diff
@@ -541,3 +554,21 @@ func (l *Log) FmtString() string {
 
 	return sb.String()
 }
+
+func runBpfProgram(prog *ebpf.Program, data, ctx []byte) (statusCode uint32, dataOut, ctxOut []byte, err error) {
```

Comment:
@YutaroHayakawa Do you think this `runBpfProgram` can help test skb->cb with reference to your PR  https://github.com/cilium/cilium/pull/24181? The parameter `ctx` will be used as skb metadata, like `memcpy(skb, ctx)`, so we can get cb by `ctx[offsetof(skb, cb)]`. In this PR we pass the returned `ctx` from the last BPF test program to the next BPF test program, so whatever changes made by the previous BPF program can be seen afterwards.

Reply:
Yeah, this helps, but I intentionally didn't use that because the `ctx_in` parameter of `BPF_PROG_TEST_RUN` is only available after kernel `5.2` and our minimal possible kernel is `4.19`. However, if we can make sure we only run these tests on kernel `5.2` or later, then it's fine. Have you discussed this with someone?